test/SparqlAntlrParserTestHelpers.h

```diff
@@ -874,19 +874,119 @@ inline auto VisibleVariables =
   return AD_PROPERTY(ParsedQuery, getVisibleVariables, testing::Eq(elems));
 };
 
+using namespace updateClause;
+
 inline auto UpdateQuery =
-    [](const std::vector<SparqlTripleSimple>& toDelete,
-       const std::vector<SparqlTripleSimple>& toInsert,
+    [](const std::vector<SparqlTripleSimpleWithGraph>& toDelete,
+       const std::vector<SparqlTripleSimpleWithGraph>& toInsert,
        const Matcher<const p::GraphPattern&>& graphPatternMatcher)
     -> Matcher<const ::ParsedQuery&> {
   return testing::AllOf(
       AD_PROPERTY(ParsedQuery, hasUpdateClause, testing::IsTrue()),
       AD_PROPERTY(ParsedQuery, updateClause,
-                  AD_FIELD(parsedQuery::UpdateClause, toDelete_,
-                           testing::ElementsAreArray(toDelete))),
+                  AD_FIELD(parsedQuery::UpdateClause, op_,
+                           testing::VariantWith<::updateClause::GraphUpdate>(
+                               AD_FIELD(GraphUpdate, toDelete_,
+                                        testing::ElementsAreArray(toDelete))))),
```

Comment:
You seem to have some duplication for the updateClause->Variant-Wityh etc. case.
I think you can use `testing::AllOf(testToDelete, testToInsert)`

Reply:
Or even better, further below you have a `GraphUpdate` matcher, can't you use that one here inside the variantWith?
(given that you declare or define it upfront?